src/bin/drag_and_drop_listbox.rs

```diff
@@ -0,0 +1,193 @@
+// A drag-and-drop example. You can drag the list items using the icon on the left, and drop them
+// to other icons to move the list item. Originally listbox-dnd.c in GTK+ repository. Ported to
+// gtk-rs by Ömer Sinan Ağacan <omeragacan@gmail.com>.
+
+extern crate cairo;
+extern crate gdk;
+extern crate gio;
+extern crate gtk;
+
+use gdk::{Atom, DragAction, DragContext, ModifierType, Screen};
+use gio::prelude::*;
+use gtk::*;
+use std::env::args;
+
+fn main() {
+    let app = gtk::Application::new("com.github.gtk-rs.examples.basic", Default::default())
+        .expect("Initialization failed...");
+
+    app.connect_activate(|app| {
+        build_ui(app);
+    });
+
+    app.run(&args().collect::<Vec<_>>());
+}
+
+fn build_ui(app: &gtk::Application) {
+    let css_provider = CssProvider::new();
+    css_provider.load_from_data(CSS.as_bytes()).unwrap();
+    StyleContext::add_provider_for_screen(
+        Screen::get_default().as_ref().unwrap(),
+        &css_provider,
+        STYLE_PROVIDER_PRIORITY_APPLICATION,
+    );
+
+    let window = gtk::ApplicationWindow::new(app);
+    window.set_title("A drag-and-drop example");
+    window.set_default_size(350, 500);
+
+    let list_box = ListBox::new();
+    list_box.set_selection_mode(SelectionMode::None);
+
+    window.add(&list_box);
+
+    // Add rows
+    for i in 0..20 {
+        list_box.add(&create_row(i));
+    }
+
+    window.show_all();
+}
+
+fn create_row(i: i32) -> ListBoxRow {
+    let list_box_row = ListBoxRow::new();
+    let box_ = Box::new(Orientation::Horizontal, 5);
+    box_.set_margin_start(10);
+    box_.set_margin_end(10);
+    list_box_row.add(&box_);
+
+    let event_box = EventBox::new();
+    let image = Image::new_from_icon_name("open-menu-symbolic", IconSize::Button);
+    event_box.add(&image);
+    box_.add(&event_box);
+
+    let label_text = format!("Row {}", i);
+    let label = Label::new(Some(label_text.as_str()));
+
+    box_.add(&label);
+
+    // Drag stuff
+    //
+    // Make the event box draggable. Triggered with left mouse button.
+    //
+    let drag_targets: [TargetEntry; 1] = [TargetEntry::new(
+        "GTK_LIST_BOX_ROW", // I don't know how this string is used
+        TargetFlags::SAME_APP,
```

Comment:
This one here is btw the reason why putting pointers into the selection data is a very bad idea (even if you're careful to get the memory management right). Things could be shared between different processes.